python_modules/dagster/dagster/_daemon/sensor.py

```diff
@@ -427,6 +448,67 @@ def _process_tick(
     )
 
 
+def _get_evaluation_tick(
+    instance: DagsterInstance,
+    sensor: ExternalSensor,
+    evaluation_timestamp: float,
+    logger: logging.Logger,
+) -> InstigatorTick:
+    """Returns the current tick that the sensor should evaluate for. If there is unfinished work
+    from the previous tick that must be resolved before proceeding, will return that previous tick.
+    """
+    origin_id = sensor.get_external_origin_id()
+    selector_id = sensor.get_external_origin().get_selector().get_id()
+    previous_tick = next(iter(instance.get_ticks(origin_id, selector_id, limit=1)), None)
+
+    # check for unfinished work on the previous tick
+    if previous_tick is not None:
+        has_unrequested_runs = len(previous_tick.unsubmitted_run_ids_with_requests) > 0
+        if previous_tick.status == TickStatus.STARTED:
+            # if the previous tick was interrupted before it was able to request all of its runs,
+            # and it hasn't been too long, then resume execution of that tick
+            if (
+                evaluation_timestamp - previous_tick.timestamp <= MAX_TIME_TO_RESUME_TICK_SECONDS
+                and has_unrequested_runs
+            ):
+                logger.warn(
+                    f"Tick {previous_tick.tick_id} was interrupted part-way through, resuming"
+                )
+                return previous_tick
+            else:
+                # previous tick won't be resumed - move it into a SKIPPED state so it isn't left
+                # dangling in STARTED, but don't return it
+                logger.warn(f"Moving dangling STARTED tick {previous_tick.tick_id} into SKIPPED")
+                previous_tick = previous_tick.with_status(status=TickStatus.SKIPPED)
+                instance.update_tick(previous_tick)
```

Comment:
ah this will be very nice (assuming we satisfy ourselves that the perf is not a concern with adding the additional fetch) - users semi-regularly report confusion about these dangling ticks